[imgui] Set C++ 11 standard in CMakeLists.txt

xaedes opened the pull request:
`imgui` code uses C++ 11 features but does not have C++ 11 enabled for compiling.
This results in compilation error, for example on Ubuntu Xenial.
To fix this the C++ 11 standard is set in the port CMakeLists.txt

- #### What does your PR fix?
  Fixes #25588

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  I think all triplets are supported. No I dont update CI baseline.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes.

xaedes committed:
[imgui] Enable C++ 11 standard using target_compile_features

As lerppana suggested:
It is better to use cxx_std_11 meta feature instead of locking the version number.
This should indicate cmake to use at least 11 instead of exactly 11.

see vcpkg issue microsoft#25588

xaedes committed:
[imgui] Enable C++ 11 standard as code requires it

see vcpkg issue microsoft#25588

Thomas1664 commented on versions/i-/imgui.json:
```diff
@@ -1,9 +1,9 @@
     {
-      "git-tree": "c28ebbdbe22a87ce01c3b2b6c15bed036721c6a0",
+      "git-tree": "a79e0e3367d45175ad025d0ed40f90d9972c2196",
       "version": "1.88",
-      "port-version": 0
+      "port-version": 1
```
Please don't modify an existing version. Revert this and run `vcpkg x-add-version imgui` instead. Then commit the result.

xaedes replied:
Oh, of course!

xaedes committed:
[imgui] update versions (vcpkg x-add-version imgui)

Enable C++ 11 standard as code requires it

see vcpkg issue microsoft#25588

xaedes commented:
I reverted the versions modifying commit.
As in the issue thread #25588 discussed, I added another commit to use cxx_std_11 meta feature instead of locking the version number. This should indicate cmake to use at least 11 instead of exactly 11.
I then commited the results from `vcpkg x-add-version imgui`
